packages/trpc/server/routers/viewer/workflows/util.ts

```diff
@@ -99,16 +95,80 @@ export const verifyEmailSender = async (email: string, userId: number, teamId: n
           teamId,
         },
       });
-    } else if (verifiedEmail.teamId !== teamId) {
+    } else if (teamId && verifiedEmail.teamId !== teamId) {
+      await prisma.verifiedEmail.create({
+        data: {
+          email,
+          userId,
+          teamId,
+        },
+      });
+    }
+    return;
+  }
+
+  const userEmail = await prisma.user.findFirst({
```

Comment:
if we didn't find `verifiedEmail` in the db, check if it's the user's email and later if it's one of the team members email